src/preferences/configobject.h

```diff
@@ -2,24 +2,42 @@
 
 #include <QDomNode>
 #include <QHash>
+#include <QJSValue>
 #include <QKeySequence>
 #include <QMap>
 #include <QMetaType>
 #include <QReadWriteLock>
 #include <QString>
+#include <QUuid>
+#include <utility>
 
+#include "controllers/scripting/legacy/controllerscriptenginelegacy.h"
 #include "util/assert.h"
 #include "util/compatibility/qhash.h"
 #include "util/debug.h"
 
+class InputHandlerExecutable {
+  public:
+    InputHandlerExecutable() = default;
+    explicit InputHandlerExecutable(QString group)
+            : group(std::move(group)) {
+    }
+    QString group;
+    QJSValue getFunction(ControllerScriptEngineLegacy& engine) const {
+        return QJSValue(QJSValue::UndefinedValue);
+    }
+    QString getFunctionDesc() const {
+        return "";
+    }
+};
+
 // Class for the key for a specific configuration element. A key consists of a
 // group and an item.
-class ConfigKey final {
+class ConfigKey final : public InputHandlerExecutable {
```

Comment:
`ConfigKey` is almost a vocabulary type used in mixxx. Its very primitive and used everywhere (in the config, the interaction with the audio engine, skins, UI, controllers, etc). In order for those different systems to stay separate, we can't mix anything (in this case InputHandlerExecutable) into it. This feature can/should be implemented entirely without touching `configobject.h`